thirdparty/lpeg/CMakeLists.txt

```diff
@@ -9,9 +9,10 @@ enable_language(C)
 assert_var_defined(CC)
 assert_var_defined(DYNLIB_CFLAGS)
 assert_var_defined(LUA_DIR)
-assert_var_defined(LUAJIT_LIB)
 
-set(BUILD_CMD_STR "${CC} ${DYNLIB_CFLAGS} -I${LUA_DIR}/src ${LUAJIT_LIB} -o lpeg.so lpcap.c lpcode.c lpprint.c lptree.c lpvm.c")
+# assuming lpeg.so is installed under $(OUTPUT_DIR)/rocks/lib/lua/5.1 and
+# libluajit.so is in $(OUTPUT_DIR)/libs
+set(BUILD_CMD_STR "${CC} ${DYNLIB_CFLAGS} -I${LUA_DIR}/src -Wl,-rpath,'../../../../libs' -lluajit -o lpeg.so lpcap.c lpcode.c lpprint.c lptree.c lpvm.c")
```

Comment:
Right... I forgot we have our own custom loader.. I can confirm that `-lluajit` is what actually fixes the crash. Without this, the custom loader will skip `libluajit.so`.

I was under the impression that RPATH is the path relative to the binary that first ran by user (in our case luajit). So looks like I am wrong and it should be the path for the dynamic libary that passed in to dlopen. That means for all libraries in `libs` dir, the correct RPATH should be `$ORIGIN` instead of `$ORIGIN/libs`?

I am confused because some of our RPATH are set to `libs` like filemq, libzyre etc. Shouldn't they be set to `$ORIGIN` too?